app/components/common/passwordInput/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+import { EyeIcon, EyeSlashIcon } from '@heroicons/react/24/outline';
+interface PasswordInputProps {
+  label: string;
+  description: string;
+}
+
+const PasswordInput: React.FC<PasswordInputProps> = ({ label, description }) => {
+  const [showPassword, setShowPasword] = useState(false);
+  const Icon = showPassword ? EyeIcon : EyeSlashIcon;
+  return (
+    <div>
+      <label>{label}</label>
+      <div className="relative">
+        <input
+          aria-describedby="desc"
+          type={showPassword ? 'text' : 'password'}
+          data-testid="password-input"
+          className={` bg-stone-50 text-sm p-3  focus:outline-none rounded-lg w-full  border-solid border  border-stone-400`}
+        />
+        <Icon
+          data-testid="icon"
+          className="h-6 w-6 absolute right-3 top-1/2 -translate-y-1/2"
+          onClick={() => setShowPasword((prev) => !prev)}
+        />
+      </div>
+      <span id="desc" className="text-stone-500 text-sm">
+        {description}
+      </span>
+    </div>
+  );
+};
+export default PasswordInput;
```

Comment:
can we use userInput for the whole component, except we could pass , `type=password` as prop and based on that we could toggle the **eyeIcon**, let me know wdyt?

Reply:
I thought it keep the logic for userInput separate but if you feel we should merge them, then sure I can do that.

Reply:
yes, lets do it if it is feasible

Reply:
This seems to be a good point as we could easily reuse the UserInput that is already accessible to us

Reply:
I tried and didn't find it much of an advantage to merging the 2 components, keeping the logic and tests separate makes it easy to use

Reply:
I agree with @isVivek99 and @harshith-venkatesh . This can be directly handled inside `userInput` as it is a type of `input`.